### PR TITLE
PHOENIX-3499 Enable null value for quote character for CSVBulkLoad tool

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/CsvBulkImportUtil.java [hunk 1]
```diff
@@ -69,8 +69,11 @@ public static void configurePreUpsertProcessor(Configuration conf,
     }
 
     @VisibleForTesting
-    static void setChar(Configuration conf, String confKey, char charValue) {
-        conf.set(confKey, Bytes.toString(Base64.getEncoder().encode(Character.toString(charValue).getBytes())));
+
+    static void setChar(Configuration conf, String confKey, Character charValue) {
+        if(charValue!=null) {
```
Comment (dbwong) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/CsvBulkImportUtil.java:74]: Can you explain the reason for the null check?

**Review by dbwong — COMMENTED** [review]
Looks okay to me, one question below.

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/CsvBulkImportUtil.java (hunk 1 above)
Comment (stoty) [reply phoenix-core/src/main/java/org/apache/phoenix/mapreduce/CsvBulkImportUtil.java:74]: org.apache.phoenix.mapreduce.CsvBulkImportUtil.getCharacter(Configuration, String) checks to see if the property is set, and will default to null, when it isn't.
It is a very circumspect way of passing a null value, but that's how the existing code works.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 44s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m  3s |  master passed  |
| +1 :green_heart: |  compile  |   1m  7s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 54s |  master passed  |
| +0 :ok: |  spotbugs  |   3m 39s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  6s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 34s |  phoenix-core: The patch generated 27 new + 78 unchanged - 17 fixed = 105 total (was 95)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 54s |  the patch passed  |
| -1 :x: |  spotbugs  |   4m 25s |  phoenix-core generated 1 new + 972 unchanged - 1 fixed = 973 total (was 973)  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 114m  8s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   | 159m  4s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:phoenix-core |
|  |  Found reliance on default encoding in org.apache.phoenix.mapreduce.CsvBulkImportUtil.setChar(Configuration, String, Character):in org.apache.phoenix.mapreduce.CsvBulkImportUtil.setChar(Configuration, String, Character): String.getBytes()  At CsvBulkImportUtil.java:[line 75] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1065 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 7f918ab5a181 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 8333ae5 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/artifact/yetus-general-check/output/new-spotbugs-phoenix-core.html |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/testReport/ |
| Max. process+thread count | 6424 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1065/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**